### DISCUSSION TITLE
Adding OpenXR Hand Interaction Profile

### DISCUSSION THREAD

**Discussion by StephenHodgson** [leapmotion/UnityModules#1283] [Ideas]
First off, props to you guys for getting a more universal hand input implementation in OpenXR done before Unity.
We've been waiting on them since the introduction of the OpenXR package. 

Just one small improvement and suggestion is to add an XRController implementation for the hands similar to how the HoloLens hand interaction profile is setup. 

This would make detection of the hands fit within the new Unity Input System and give us the ability to listen for gestures similar to that HoloLens hand interaction profile input scheme.

![image](https://user-images.githubusercontent.com/13334553/186169032-664d013d-e616-4043-9061-54a5073389ec.png)

**Comment by StephenHodgson** [comment]
Also is the source for [UltraleapOpenXRUnity.dll](https://github.com/ultraleap/UnityPlugin/blob/develop/Packages/Tracking%20OpenXR/Runtime/Plugins/x64/UltraleapOpenXRUnity.dll) also under MIT license?

I was hoping I could take a stab at contributing to this myself. We may have to add to the xr input subsystem to properly wire up inputs to the input system.

**Reply by MattGrayUL** [reply]
Whilst in pre-release we have been developing this layer privately. We may open it up in the future but we need to consider that the OpenXR API will need to be kept aligned

Great to know there is interest though, I will let the team know 🙂

**Reply by StephenHodgson** [reply]
Without the interaction profile, the implementation feels incomplete, tbh. 

This would be great to get proper functionality with the input system similar to the other hand interaction profiles

**Comment by MattGrayUL** [comment]
We have plans to add an Interaction Profile for OpenXR soon and some of the preparation is already complete

Your suggestion will help with increasing the known desire for such a feature. Thanks 🙂
